src/internal/single-tab-stop/__tests__/utils.tsx

```diff
@@ -45,24 +61,32 @@ const FakeSingleTabStopNavigationProvider = forwardRef(
   }
 );
 
+export function getTestSingleTabStopNavigationProvider(providerId?: string): TestProviderAPI {
+  return {
+    setCurrentTarget(focusTarget, suppressed) {
+      if (providerId) {
+        providerRegistry.get(providerId)?.setCurrentTarget(focusTarget, suppressed);
+      } else {
+        Array.from(providerRegistry).forEach(([, provider]) => provider.setCurrentTarget(focusTarget, suppressed));
+      }
+    },
+  };
+}
+
+/**
+ * @deprecated - Use TestSingleTabStopNavigationProvider instead
+ */
 export function renderWithSingleTabStopNavigation(
```

Comment:
Can you re-write [this test-case](https://github.com/cloudscape-design/component-toolkit/blob/1bdc64ff9606237576e0b4ea78057ae77bf73ac2/src/internal/single-tab-stop/__tests__/context.test.tsx#L59) to a non-deprecated version to prove that it works?

Reply:
I updated two tests to make sure both the provider and the helper util work as expected. Once components code is migrated, I will remove the renderWithSingleTabStopNavigation and migrate the rest of the tests.